src/javascripts/button_state.js

```diff
@@ -0,0 +1,35 @@
+// Adds 'loading icon' to button with id of 'btn__loading' when clicked
+
+(function () {
+	var btnLoad, input;
+
+	// Grab elements with id of btn__loading
+	btnLoad = document.getElementById('btn_loading');
+
+	// Grab elements with id of search_box
+	input = document.getElementById('input');
+
+	// Only create the event listener if the element exists within DOM
+	if (typeof (btnLoad) != 'undefined' && btnLoad != null) {
```

Comment:
If the button is `false, 0, NaN, or ""`, btnLoad won't be undefined or null, which will execute this function unnecessarily. It might be worth just using `if(btnLoad)`.